src/common/src/hash/table_distribution.rs

```diff
@@ -13,30 +13,35 @@
 // limitations under the License.
 
 use std::mem::replace;
-use std::ops::Deref;
 use std::sync::{Arc, LazyLock};
 
 use itertools::Itertools;
 use risingwave_pb::plan_common::StorageTableDesc;
-use tracing::warn;
 
 use crate::array::{Array, DataChunk, PrimitiveArray};
-use crate::bitmap::{Bitmap, BitmapBuilder};
+use crate::bitmap::Bitmap;
 use crate::hash::VirtualNode;
 use crate::row::Row;
 use crate::util::iter_util::ZipEqFast;
 
 /// For tables without distribution (singleton), the `DEFAULT_VNODE` is encoded.
 pub const DEFAULT_VNODE: VirtualNode = VirtualNode::ZERO;
+pub use DEFAULT_VNODE as SINGLETON_VNODE;
+
+use super::VnodeBitmapExt;
 
 #[derive(Debug, Clone)]
 enum ComputeVnode {
```

Comment:
May just rename it to `TableDistribution`, because the `TableDistribution` only wraps `compute_vnode` now.

Reply:
Not sure whether we could introduce more common fields in the future. Up to @wenym1 to decide.